paddle/fluid/operators/activation_op.cc

```diff
@@ -695,6 +702,8 @@ class LeakyReluDoubleGradMaker
     op->SetType("leaky_relu_grad_grad");
     // input1: X
     op->SetInput("X", Input("X"));
+    // input2: Out
+    op->SetInput("Out", Input("Out"));
```

Comment:
Why add an Input here?

Reply:
It's because MKLDNN activation function grad functor expects to have input called "Out" to create hash key for caching. Without this change activation grad test failed.

Reply:
> create hash key for caching

Why should we add this input only for creating hash key, since this op doesn't need this input indeed?

Reply:
Well, I did it as it felt like the fastest way to get around this error but you are right, this Input unneccesary so I will change the way MKLDNN key is generated.

Reply:
Thanks very much! If we change the input, then many trained models before will fail after update the develop branch.

Reply:
@luotao1 Could you please explain why trained model will fail when we add additional input to grad op? This is just leaky relu grad op , so this grad op is not holding any params, so why extending inputs for grad op would cause failure? I'm asking because as described in #17960 I would like to extend all mkl-dnn grad ops to have X as input of grad ops as mkl-dnn backward implementations are requiring X to compute backward/grad.

Reply:
@phlrain Could you help see this problem?

Reply:
@jczaja @grygielski 
Discussed with @phlrain, it's OK to add additional input to grad op as this PR does. But we could not `addinput()` in `OpMaker()`.

Reply:
Ok, so my sollution is acceptable? Because during key generation we would have to add another cases to check if specific operator grad has only X input what causes additional problems when adding MKLDNN support for further operatiors.

Reply:
Yes, it is.